src/pybind/mgr/cephadm/services/cephadmservice.py

```diff
@@ -426,6 +431,77 @@ def create(self, daemon_spec: CephadmDaemonSpec) -> str:
 
         return self.mgr._create_daemon(daemon_spec)
 
+    def create_realm_zonegroup_zone(self, spec: RGWSpec, rgw_id: str):
+        # check cluster health
+        ret, out, err = self.mgr.check_mon_command({
+            'prefix': 'health',
+            'format': 'json',
+        })
+        try:
+            j = json.loads(out)
+            self.mgr.log.info(j)
+        except Exception as e:
+            raise OrchestratorError('failed to parse health status')
+
+        if j['status'] != 'HEALTH_OK':
+                raise OrchestratorError('Health not ok, will try agin when health ok')
+
+        #get keyring needed to run rados commands
+        ret, keyring, err = self.mgr.check_mon_command({
+            'prefix': 'auth get-or-create',
+            'entity': f"{utils.name_to_config_section('rgw')}.{rgw_id}",
+            'caps': ['mon', 'allow *',
+                     'mgr', 'allow rw',
+                     'osd', 'allow rwx'],
+        })
+        #strip out just the keyring
+        keyring = keyring.split('key = ',1)[1].rstrip()
+
+        # get realms
+        result = subprocess.run(['radosgw-admin', '--key=%s'%keyring, '--user', 'rgw.%s'%rgw_id, 'realm', 'list', '--format=json'], stdout=subprocess.PIPE, stderr=subprocess.PIPE)
```

Comment:
do you want to call `subprocess.check_output` here?

Reply:
I chose to use run() instead of check_output() because check_output raises a CalledProcessError if the return code is not 0. when calling 'radosgw-admin realm list' without any realms existing and no rgw service running you get the following and a return code of 2. a similar thing happens for zonegroups and zones.

```
[ceph: root@vm-00 /]# radosgw-admin realm list 
failed to list realms: (2) No such file or directory
[ceph: root@vm-00 /]# echo $?
2
```

I could have used check_output with some try catch blocks but i thought the code looked cleaner this way.